django/db/models/fields/related.py

```diff
@@ -613,6 +613,17 @@ def resolve_related_fields(self):
             )
             to_field = (self.remote_field.model._meta.pk if to_field_name is None
                         else self.remote_field.model._meta.get_field(to_field_name))
+            if to_field and to_field.model != self.remote_field.model._meta.concrete_model:
+                raise exceptions.FieldError(
+                    "ForeignKey '%s.%s' pointing to a non local remote field '%s.%s'." %
+                    (
+                        self.model._meta.label,
+                        self.name,
+                        to_field.model._meta.label,
```

Comment:
This should be `self.remote_field.model._meta.concrete_model._meta.label` as [pointed by Simon](https://github.com/django/django/pull/12383#discussion_r372421836).